File: EuclidGCDAlgorithm/EuclidGCD.go
package main

import "fmt"

func swap(firstNum int, secondNum int) (int, int) {
	return secondNum, firstNum
}

func getGCD(firstNum int, secondNum int) int {
	if firstNum < secondNum {
		firstNum, secondNum = swap(firstNum, secondNum)
	}

	if secondNum > 0 {
		return getGCD(secondNum, firstNum%secondNum)
	} else {
		return firstNum
	}
}

func main() {
	fmt.Println("GCD(6, 4) => ", getGCD(6, 4))
	fmt.Println("GCD(4, 6) => ", getGCD(4, 6))
	fmt.Println("GCD(2024, 125) => ", getGCD(2024, 125))
}

File: EuclidGCDAlgorithm/EuclidGCD.java
public class EuclidGCD {
    
    static int getGCD(int numerator, int denominator) {

        //Swap if numerator is smaller than denominator
        if(denominator > numerator){
            int temp = numerator;
            numerator = denominator;
            denominator = temp;
        }

        while (denominator > 0) {
            int reminder = numerator % denominator;
            numerator = denominator;
            denominator = reminder;
        }

        return numerator;
    }

    public static void main(String[] args) {
        System.out.println("Running GCD TEST");
        System.out.println("GCD(6,4) => " + EuclidGCD.getGCD(6,4));
        System.out.println("GCD(4,6) => " + EuclidGCD.getGCD(4,6));
        System.out.println("GCD(2024, 125) => "+ EuclidGCD.getGCD(2024, 125));
    }
}